lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -41,17 +51,20 @@ type ImageCache struct {
 	cache map[string]*metadata.ImageConfig
 }
 
+// NewImageCache creates and returns a new ImageCache
 func NewImageCache() *ImageCache {
 	return &ImageCache{
 		cache: make(map[string]*metadata.ImageConfig),
 	}
 }
 
-// Update adds new layers to the cache
+// Update adds new images to the cache
 func (c *ImageCache) Update(client *client.PortLayer) error {
 	c.m.Lock()
 	defer c.m.Unlock()
 
+	CacheNotUpdated = true
```

Comment:
You should also initialize this in the var section above or put in init() just to avoid any potential for race condition where we try to get before update.  It'll only happen in the event that there is a change in how we rehydrate in the future.  Never know.